lms/djangoapps/completion/api/v1/tests/test_views.py

```diff
@@ -1,139 +1,190 @@
+# -*- coding: utf-8 -*-
 """
 Test models, managers, and validators.
 """
 
-from __future__ import absolute_import, division, print_function, unicode_literals
-
+import ddt
 from django.core.urlresolvers import reverse
-from django.test import TestCase
-
-from rest_framework.test import APIRequestFactory
-from rest_framework.test import force_authenticate
+from rest_framework.test import APIClient, force_authenticate
 
+from completion import waffle
 from student.tests.factories import UserFactory, CourseEnrollmentFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+from openedx.core.djangoapps.content.course_structures.tasks import update_course_structure
 
-from lms.djangoapps.completion.api.v1.views import CompletionBatchView
-from lms.djangoapps.completion import waffle
 
-
-class CompletionBatchTestCase(TestCase):
+@ddt.ddt
+class CompletionBatchTestCase(ModuleStoreTestCase):
     """
     Test that BlockCompletion.objects.submit_batch_completion has the desired
     semantics.
     """
-    @classmethod
-    def setUpClass(cls):
-        super(CompletionBatchTestCase, cls).setUpClass()
-        cls.url = reverse('completion_api:v1:completion-batch')
-        cls.view = CompletionBatchView()
-        cls.factory = APIRequestFactory()
-        cls.user = UserFactory.create()
-        cls.course_key = 'course-v1:edx+test+run'
+    ENROLLED_USERNAME = 'test_user'
+    UNENROLLED_USERNAME = 'unenrolled_user'
+    COURSE_KEY = 'TestX/101/Test'
+    BLOCK_KEY = 'i4x://TestX/101/problem/Test_Problem'
 
     def setUp(self):
+        """
+        Create the test data.
+        """
+        super(CompletionBatchTestCase, self).setUp()
+        self.url = reverse('completion_api:v1:completion-batch')
+
+        # Enable the waffle flag for all tests
         _overrider = waffle.waffle().override(waffle.ENABLE_COMPLETION_TRACKING, True)
         _overrider.__enter__()
         self.addCleanup(_overrider.__exit__, None, None, None)
 
-    def enroll(self):
-        CourseEnrollmentFactory.create(user=self.user, course_id=self.course_key)
+        # Create course
+        self.course = CourseFactory.create(org='TestX', number='101', display_name='Test')
+        self.problem = ItemFactory.create(
+            parent=self.course,
+            category="problem",
+            display_name="Test Problem",
+        )
+        update_course_structure(unicode(self.course.id))
```

Comment:
@macornwell This was the magic line that creates the CourseStructure objects.  Took some finding..

Reply:
Great! Seems like something that should be auto-hooked up somewhere in a model save.